asab/api/doc.py

```diff
@@ -7,54 +7,143 @@
 import aiohttp.web
 import yaml
 
+from .doc_templates import SWAGGER_OAUTH_PAGE, SWAGGER_DOC_PAGE
+
 ##
 
 L = logging.getLogger(__name__)
 
-
 ##
 
-
 class DocWebHandler(object):
-
-	def __init__(self, api_service, app, web_container, config_section_name='asab:doc'):
+	def __init__(self, api_service, app, web_container, config_section_name="asab:doc"):
 		self.App = app
 		self.WebContainer = web_container
-		self.WebContainer.WebApp.router.add_get('/doc', self.doc)
-		self.WebContainer.WebApp.router.add_get('/oauth2-redirect.html', self.oauth2_redirect)
-		self.WebContainer.WebApp.router.add_get('/asab/v1/openapi', self.openapi)
+		self.WebContainer.WebApp.router.add_get("/doc", self.doc)
+		self.WebContainer.WebApp.router.add_get(
+			"/oauth2-redirect.html", self.oauth2_redirect
+		)
+		self.WebContainer.WebApp.router.add_get("/asab/v1/openapi", self.openapi)
 
-		self.AuthorizationUrl = asab.Config.get(config_section_name, "authorizationUrl", fallback=None)
+		self.AuthorizationUrl = asab.Config.get(
+			config_section_name, "authorizationUrl", fallback=None
+		)
 		self.TokenUrl = asab.Config.get(config_section_name, "tokenUrl", fallback=None)
 		self.Scopes = asab.Config.get(config_section_name, "scopes", fallback=None)
 
 		self.Manifest = api_service.Manifest
 
+	def build_swagger_documentation(self) -> dict:
+		"""Take a docstring of the class and a docstring of methods and merge them into Swagger data."""
+		specification: dict = {}
+		# TODO: check for the add_dict in the primary code
+		app_doc_string: str = self.App.__doc__
+		additional_info_dict: dict = {}
+
+		app_description: str = self.get_description(app_doc_string)
+		additional_info_dict.update(self.get_additional_info(app_doc_string))
+		if additional_info_dict is not None:
+			specification.update(additional_info_dict)
+
+		specification.update(self.prepare_description_frame(app_description))
+		specification["components"] = self.create_security_schemes()
+		specification["info"]["version"] = self.get_manifest()
+		specification["info"]["description"] = self.get_server_and_container_info(
+			app_description
+		)
 
-	def build_swagger_specs(self):
-		"""
-		Takes a docstring of a class and a docstring of methods and merges them
-		into a Swagger specification.
-		"""
+		# routers sorting
+		asab_routers = []
+		doc_routers = []
+		microservice_routers = []
```

Comment:
"routes", not "routers".

also you probably don't need three lists. doc_routes are a subset of asab_routes and we don't need to differentiate between them as long as they come after microservice_routes.